plugins/device/transaction.go

```diff
@@ -1,52 +1,50 @@
 package device
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/gluster/glusterd2/glusterd2/transaction"
 	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
-
-	"github.com/pborman/uuid"
-	log "github.com/sirupsen/logrus"
+	"github.com/gluster/glusterd2/plugins/device/deviceutils"
 )
 
 func txnPrepareDevice(c transaction.TxnCtx) error {
-	var peerID uuid.UUID
-	var req deviceapi.AddDeviceReq
-	var deviceList []deviceapi.Info
-	if err := c.Get("peerid", peerID); err != nil {
-		c.Logger().WithError(err).Error("Failed transaction, cannot find peer-id")
+	var peerID string
+	if err := c.Get("peerid", &peerID); err != nil {
+		c.Logger().WithError(err).WithField("key", "peerid").Error("Failed to get key from transaction context")
 		return err
 	}
-	if err := c.Get("req", req); err != nil {
-		c.Logger().WithError(err).Error("Failed transaction, cannot find device-details")
+
+	var device string
+	if err := c.Get("device", &device); err != nil {
+		c.Logger().WithError(err).WithField("key", "device").Error("Failed to get key from transaction context")
 		return err
 	}
-	for _, name := range req.Devices {
-		tempDevice := deviceapi.Info{
-			Name: name,
-		}
-		deviceList = append(deviceList, tempDevice)
+
+	var deviceInfo deviceapi.Info
+	deviceInfo.Name = device
+
+	err := deviceutils.CreatePV(device)
+	if err != nil {
+		c.Logger().WithError(err).WithField("device", device).Error("Failed to create physical volume")
+		return err
 	}
-	for index, element := range deviceList {
-		pvcreateCmd := exec.Command("pvcreate", "--metadatasize=128M", "--dataalignment=256K", element.Name)
-		if err := pvcreateCmd.Run(); err != nil {
-			c.Logger().WithError(err).WithField("device", element.Name).Error("pvcreate failed for device")
-			deviceList[index].State = deviceapi.DeviceFailed
-			continue
-		}
-		vgcreateCmd := exec.Command("vgcreate", strings.Replace("vg"+element.Name, "/", "-", -1), element.Name)
-		if err := vgcreateCmd.Run(); err != nil {
-			c.Logger().WithError(err).WithField("device", element.Name).Error("vgcreate failed for device")
-			deviceList[index].State = deviceapi.DeviceFailed
-			continue
+	vgName := strings.Replace("vg"+device, "/", "-", -1)
+	err = deviceutils.CreateVG(device, vgName)
+	if err != nil {
+		c.Logger().WithError(err).WithField("device", device).Error("Failed to create volume group")
+		errPV := deviceutils.RemovePV(device)
+		if errPV != nil {
+			c.Logger().WithError(err).WithField("device", device).Error("Failed to remove physical volume")
 		}
-		deviceList[index].State = deviceapi.DeviceEnabled
+		return err
 	}
-	err := AddDevices(deviceList, peerID.String())
+	c.Logger().WithField("device", device).Info("Device setup successful, setting device status to 'Enabled'")
+	deviceInfo.State = deviceapi.DeviceEnabled
```

Comment:
You can create the entire `deviceapi.Info` structure here itself for better readability: 

```go
deviceInfo = deviceapi.Info {
    Name : device,
    State: deviceapi.DeviceEnabled,
}
```